cdap-app-fabric/src/main/java/io/cdap/cdap/internal/provision/DefaultProvisionerContext.java

```diff
@@ -107,4 +110,9 @@ public LocationFactory getLocationFactory() {
   public RuntimeMonitorType getRuntimeMonitorType() {
     return runtimeMonitorType;
   }
+
+  @Override
+  public MetricsContext getMetricsContext() {
```

Comment:
Shouldn't expose the raw `MetricsContext`. Whatever emitted from the provisioner should already have the current metrics context setup. Also, since MetricsContext is in cdap-watchdog-api, which is not a dependency of cdap-runtime-spi, you should either have the emit metric method defined in the ProvisionerContext, or have a different metrics context defined in the cdap-runtime-spi module to better suit the provisioner SPI need.

Reply:
fixed